integration/src/main/java/scripts/testscripts/ListWorkspaces.java

```diff
@@ -0,0 +1,93 @@
+package scripts.testscripts;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import bio.terra.workspace.api.WorkspaceApi;
+import bio.terra.workspace.model.WorkspaceDescription;
+import bio.terra.workspace.model.WorkspaceDescriptionList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import scripts.utils.ClientTestUtils;
+import scripts.utils.WorkspaceAllocateTestScriptBase;
+
+/**
+ * This test exercises WSM's listWorkspaces endpoint. Note that because this test reads the full
+ * set of a user's workspaces, it is not thread safe and only one instance of this test should
+ * run against a server at once.
+ */
+public class ListWorkspaces extends WorkspaceAllocateTestScriptBase {
+
+  private UUID workspaceId2;
+  private UUID workspaceId3;
+  private WorkspaceApi secondUserApi;
+
+  @Override
+  protected void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi) throws Exception {
+    assertThat("There must be two test users to run this test",
+        testUsers.size() == 2);
+    // Primary test user (index 0) is handled in the test base.
+    TestUserSpecification nonWorkspaceUser = testUsers.get(1);
+    secondUserApi = ClientTestUtils.getWorkspaceClient(nonWorkspaceUser, server);
+    // We expect test users to never have permanent workspaces. However, they may have some
+    // workspaces due to previous test failures. Before running this test, we clean up all
+    // workspaces for both test users.
+    WorkspaceDescriptionList preTestWorkspaceList = workspaceApi.listWorkspaces(/*offset=*/0, /*limit=*/10000);
+    for (WorkspaceDescription workspace : preTestWorkspaceList.getWorkspaces()) {
```

Comment:
I think we need new test users for this test so there's no interference with other test runs. As written, not only is it not safe to run two instances of this test at once, but it's also possible to kill workspaces needed by someone else running integration tests in the environment.

It would be handy to have the ability to keep a test workspace around for investigation in the cloud console, but that's not possible with Janitor as I understand it.

Reply:
Yeah, that's a good point about multiple people running tests at once. Even with separate test users I think this would be a hassle, since we generally run large test suites and bad timing could ruin both test runs.
Refactored to be resilient to multiple test runs, though with slightly less coverage